api/src/main/java/ai/djl/modality/audio/AudioFactory.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.modality.audio;
+
+import ai.djl.ndarray.NDArray;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * {@code AudioFactory} contains audio creation mechanism on top of different platforms like PC and
+ * Android. System will choose appropriate Factory based on the supported audio type.
+ */
+public abstract class AudioFactory {
+
+    private static final Logger logger = LoggerFactory.getLogger(AudioFactory.class);
+
+    @SuppressWarnings({"PMD.DoubleBraceInitialization", "PMD.UseConcurrentHashMap"})
```

Comment:
better avoid create a new anonymous class, you can use a static initializer instead.

This can be just a String[], no need use a map here

Reply:
OK